llvm/lib/Target/AArch64/AArch64ISelLowering.cpp

```diff
@@ -1777,10 +1777,12 @@ AArch64TargetLowering::AArch64TargetLowering(const TargetMachine &TM,
 
     // Histcnt is SVE2 only
     if (Subtarget->hasSVE2()) {
-      setOperationAction(ISD::EXPERIMENTAL_VECTOR_HISTOGRAM, MVT::Other,
+      setOperationAction(ISD::EXPERIMENTAL_VECTOR_HISTOGRAM, MVT::i8, Promote);
+      setOperationAction(ISD::EXPERIMENTAL_VECTOR_HISTOGRAM, MVT::i16, Promote);
```

Comment:
These two `setOperationActions`s are unnecessary, because i8 and i16 are illegal types, so the legalizer will already try to legalize the operation, regardless of this `setOperationAction`.